packages/shared/src/lib/core/wallet/actions/updateL2BalanceWithoutActivity.ts

```diff
@@ -4,11 +4,11 @@ import { updateLayer2AccountBalanceForTokenOnChain } from '@core/layer-2/stores'
 import { NetworkId } from '@core/network/types'
 
 export async function updateL2BalanceWithoutActivity(
-    rawAmount: number,
+    rawAmount: bigint,
     tokenId: string,
     account: IAccountState,
     networkId: NetworkId
 ): Promise<void> {
     const newBalance = updateLayer2AccountBalanceForTokenOnChain(account.index, networkId, tokenId, rawAmount)
-    await calculateAndAddPersistedTokenBalanceChange(account, networkId, tokenId, newBalance, true)
+    await calculateAndAddPersistedTokenBalanceChange(account, networkId, tokenId, newBalance.toString(), true)
```

Comment:
we convert this to string but later convert it to a bigint with aditional checks? is there a reason it needs to be string here

Reply:
This is because atm token balance changes are persisted as a string. Alternatively, we can write a migration function that converts them all to BigInt's.

Reply:
I just tried this, but it doesn't work. Below is an excerpt from the code of persistent stores. You can't store anything that cannot be `JSON.stringify()`-d. 
![image](https://github.com/bloomwalletio/bloom/assets/45079109/bd01c718-d6b5-4cf5-9005-323bdc061466)